### ISSUE TITLE
[cloud_firestore] Problems when running on iOS with flutter version 1.20.2

### ISSUE THREAD

**Issue by Tedsterh** [FirebaseExtended/flutterfire#3203]
**Describe the bug**
I am having issues using cloud firestore in my project. The app runs perfectly fine on android but when I try to run on iOS I get a build failure.

Build output from the flutter run
```
Xcode build done.                                           10.1s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    Command MergeSwiftModule failed with a nonzero exit code
    /myapp/ios/Pods/gRPC-Core/src/core/ext/upb-generated/envoy/service/discovery/v2/ads.upb.c:10:10: fatal error:
    'upb/msg.h' file not found
    #include "upb/msg.h"
             ^~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: Mapping architecture armv7 to i386. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS
    Simulator platform. (in target 'in_app_purchase' from project 'Pods')
    warning: Mapping architecture arm64 to x86_64. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS
    Simulator platform. (in target 'in_app_purchase' from project 'Pods')

Could not build the application for the simulator.
Error launching application on iPhone 8 Plus.
```

**To Reproduce**
1. I started a new flutter project, and it ran on iOS
2. I then added the cloud firestore to my pubspec
`cloud_firestore: ^0.13.7`
3. Ran the pub get, and then started the app in debug

**Expected behavior**
App should start and show the default counter app

**Additional context**

**Flutter doctor**

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 1.20.2, on Mac OS X 10.15.6 19G73, locale en-GB)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
[✓] Android Studio (version 3.5)
[✓] VS Code (version 1.48.0)
[✓] Connected device (1 available)

• No issues found!
```

**Comment by kroikie** [comment]
@Tedsterh could you try one of the following suggested in the [flutterfire](https://firebase.flutter.dev/docs/migration#cocoapods-could-not-find-compatible-versions-for-pod) docs?

**Comment by Tedsterh** [comment]
Sadly still no luck, I removed the `Podfile.lock` and ran the `pod install`.
It still did not build.
So I ran `pod repo update`, but still did not work, same errors as before:

```
myapp/ios/Pods/gRPC-Core/src/core/ext/upb-generated/
    google/protobuf/wrappers.upb.c:10:10: fatal error: 'upb/msg.h' file not found
    #include "upb/msg.h"
             ^~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
```

**Comment by mimya01** [comment]
hi 
I solve it with :
`rm -rf Pods Podfile.lock`  // _iside <your_project_path>/ios_
then 
`pod cache clean --allpod`
then 
` pod install`  _// iside <your_project_path>/ios    (Don't worry if takes a lot of time)_
now you can build your app
**Good luck**

**Comment by russellwheatley** [comment]
Please file a new issue if this problem persists, this is a major version out of date for firestore and flutter.
